xunit.runner.visualstudio.desktop/VsTestRunner.cs

```diff
@@ -122,7 +143,17 @@ void ITestExecutor.RunTests(IEnumerable<string> sources, IRunContext runContext,
                                        .Where(file => !platformAssemblies.Contains(Path.GetFileName(file))));
             }
 
-            RunTests(runContext, frameworkHandle, logger, () => GetTests(sources, logger, runContext, RunSettingsHelper.DesignMode));
```

Comment:
Issue: in case of Run All, a `Discovery` was first triggered in separate appdomain, tests were serialized, `Execution` is triggered in another appdomain.

Solution: in case of Run All, delay the actual discovery. First create an appdomain (XunitFrontController), then do a discovery within that, serialization is not required (if `RequireXunitTestProperty` is false), do the execution within same appdomain.

Since it is the same assembly, discovery and execution may happen in same appdomain (similar to functionality of xunit.console).